pkg/api/v1alpha1/cluster_types.go

```diff
@@ -290,6 +291,26 @@ type Services struct {
 	CidrBlocks []string `json:"cidrBlocks,omitempty"`
 }
 
+type DNS struct {
+	// ResolvConf refers to the DNS resolver configuration
+	ResolvConf *ResolvConf `json:"resolvConf,omitempty"`
```

Comment:
Does this need to be a pointer?

Reply:
I decided to make it a pointer because that's consistent with how we pass in the fields for all of the other extraArgs. Having this as a pointer also allows us to check if it's nil before attempting to add it to the kubelet config

Reply:
Got you. In that case, do we need the check for empty string in the path field.
If the user already has a way to *not* configure this, shouldn't we just use whatever they pass? We could also add a validation that throws an error if empty string.

Reply:
I actually ended up changing it to what you originally suggested. So no pointer & we're checking if the path is an empty string before adding to extra args.
And in the design, we decided to just behave as usual if the path is an empty string so it would end up using the default resolvConf file.